apps/guardian-ui/src/admin/FederationAdmin.tsx

```diff
@@ -69,6 +72,14 @@ export const FederationAdmin: React.FC = () => {
     const offline = status?.federation ? status.federation.peers_offline : 0;
 
     const totalPeers = online + offline;
+    const onlinePercentage = online / totalPeers;
+    if (onlinePercentage === 1) {
+      setStatusColor('green');
+    } else if (onlinePercentage >= 2 / 3) {
+      setStatusColor('yellow');
+    } else {
+      setStatusColor('red');
+    }
```

Comment:
All of the `set*` calls in here should just be calculated in the render, not in the `useEffect`. That causes an unnecessary second render. It goes

1. `status` reference changes
2. render runs (with old `guardians` and `statusColor`)
3. `useEffect` runs where `set*` calls are made
4. render runs again (with updated values)

Since these are simple values where we don't need to maintain a reference, they can just be calculated and assigned each render.

Reply:
I appreciate your React framework feedback.